lib/ecto/adapters/sql.ex

```diff
@@ -1351,22 +1398,24 @@ defmodule Ecto.Adapters.SQL do
     end
   end
 
-  defp log_stacktrace(stacktrace, repo) do
-    with [_ | _] <- stacktrace,
-         {module, function, arity, info} <- last_non_ecto(Enum.reverse(stacktrace), repo, nil) do
+  defp log_stacktrace([_ | _] = stacktrace, repo, {module, function, args}) do
+    entries = apply(module, function, args ++ [Enum.reverse(stacktrace), repo])
```

Comment:
Generally speaking, we prepend to the args. Also, the need to reverse is an implementation detail of `last_non_ecto_stacktrace`. So this should be:

```suggestion
    entries = apply(module, function, [repo, stacktrace | args])
```